File: src/main/java/com/nanabell/sponge/nico/command/economy/NicoGetCommand.java
package com.nanabell.sponge.nico.command.economy;

import com.nanabell.sponge.nico.command.SelfSpecCommand;
import com.nanabell.sponge.nico.economy.NicoCurrency;
import org.jetbrains.annotations.NotNull;
import org.spongepowered.api.Sponge;
import org.spongepowered.api.command.CommandException;
import org.spongepowered.api.command.CommandResult;
import org.spongepowered.api.command.CommandSource;
import org.spongepowered.api.command.args.CommandContext;
import org.spongepowered.api.command.spec.CommandExecutor;
import org.spongepowered.api.command.spec.CommandSpec;
import org.spongepowered.api.entity.living.player.Player;
import org.spongepowered.api.service.ServiceManager;
import org.spongepowered.api.service.economy.EconomyService;
import org.spongepowered.api.service.economy.account.UniqueAccount;
import org.spongepowered.api.text.Text;

import java.util.Optional;

public class NicoGetCommand implements CommandExecutor, SelfSpecCommand {

    private final ServiceManager serviceManager = Sponge.getServiceManager();

    @Override
    public CommandResult execute(CommandSource src, CommandContext args) throws CommandException {
        if (!(src instanceof Player)) {
            throw new CommandException(Text.of("This Command can only be used by players!"));
        }

        Player player = (Player) src;

        Optional<EconomyService> oService = serviceManager.provide(EconomyService.class);
        if (!oService.isPresent()) {
            throw new CommandException(Text.of("No EconomyService registered!"));
        }

        EconomyService service = oService.get();
        NicoCurrency currency = NicoCurrency.getCurrency();
        UniqueAccount account = service.getOrCreateAccount(player.getUniqueId()).orElseThrow(() -> new CommandException(Text.of("Unable to create User Account for User" + player)));

        player.sendMessage(Text.of("You currently have ").concat(currency.format(account.getBalance(currency))));
        return CommandResult.success();
    }

    @Override
    public @NotNull String[] aliases() {
        return new String[]{"get"};
    }

    @Override
    public @NotNull CommandSpec spec() {
        return CommandSpec.builder()
                .description(Text.of("View your current Nico Points"))
                .executor(this)
                .build();
    }
}

File: src/main/kotlin/com/nanabell/sponge/nico/command/economy/NicoGetCommand.kt
package com.nanabell.sponge.nico.command.economy

import com.nanabell.sponge.nico.command.Args
import com.nanabell.sponge.nico.command.SelfSpecCommand
import com.nanabell.sponge.nico.command.requirePlayerOrArg
import com.nanabell.sponge.nico.internal.extension.orNull
import com.nanabell.sponge.nico.internal.extension.toText
import org.spongepowered.api.Sponge
import org.spongepowered.api.command.CommandException
import org.spongepowered.api.command.CommandResult
import org.spongepowered.api.command.CommandSource
import org.spongepowered.api.command.args.CommandContext
import org.spongepowered.api.command.spec.CommandExecutor
import org.spongepowered.api.command.spec.CommandSpec
import org.spongepowered.api.service.economy.EconomyService
import org.spongepowered.api.service.permission.PermissionDescription
import org.spongepowered.api.text.Text

class NicoGetCommand : CommandExecutor, SelfSpecCommand {

    private val economy = Sponge.getServiceManager().provideUnchecked(EconomyService::class.java)

    override fun aliases(): Array<String> {
        return arrayOf("get")
    }

    override fun spec(): CommandSpec {
        return CommandSpec.builder()
                .permission("nico.command.get.base")
                .description(Text.of("View your current Nico Points"))
                .arguments(Args.optional(Args.requiringPermission(Args.playerOrSource("player".toText()), "nico.command.get")))
                .executor(this)
                .build()
    }

    override fun permissionDescriptions(builder: PermissionDescription.Builder) {
        builder.id("nico.command.get.base").register()
        builder.id("nico.command.get").register()
    }

    @Throws(CommandException::class)
    override fun execute(src: CommandSource, args: CommandContext): CommandResult {
        val player = src.requirePlayerOrArg(args, "player")
        val account = economy.getOrCreateAccount(player.uniqueId).orNull()
                ?: throw CommandException("Unable to get EconomyAccount. Wrong user?".toText())

        val message = "Your Currency Report:".toText().concat(Text.NEW_LINE).toBuilder()
        economy.currencies.forEach {
            if (account.hasBalance(it)) {
                message.append(it.format(account.getBalance(it))).append(Text.NEW_LINE)
            }
        }

        src.sendMessage(message.build())
        return CommandResult.success()
    }
}